api/client/retl/types.go

```diff
@@ -0,0 +1,70 @@
+package retl
+
+import (
+	"time"
+)
+
+// State represents the complete RETL state
+type State struct {
+	Version   string                   `json:"version"`
+	Resources map[string]ResourceState `json:"resources"`
+}
+
+// ResourceState represents the state of a single RETL resource
+type ResourceState struct {
+	ID           string                 `json:"id"`
+	Type         string                 `json:"type"`
+	Input        map[string]interface{} `json:"input"`
+	Output       map[string]interface{} `json:"output"`
+	Dependencies []string               `json:"dependencies"`
+}
+
+// PutStateRequest represents the request to update a resource state
+type PutStateRequest struct {
+	ID    string        `json:"id"`
+	URN   string        `json:"urn"`
+	State ResourceState `json:"state"`
+}
+
+// RETLSource represents a RETL source in the API
+type RETLSource struct {
+	ID                   string           `json:"id,omitempty"`
+	Name                 string           `json:"name"`
+	Config               RETLSourceConfig `json:"config"`
+	IsEnabled            bool             `json:"enabled,omitempty"`
+	SourceType           string           `json:"sourceType"`
+	SourceDefinitionName string           `json:"sourceDefinitionName"`
+	AccountID            string           `json:"accountId"`
+	CreatedAt            *time.Time       `json:"createdAt,omitempty"`
+	UpdatedAt            *time.Time       `json:"updatedAt,omitempty"`
+}
+
+type RETLSourceCreateRequest struct {
+	ID                   string           `json:"id,omitempty"`
+	Name                 string           `json:"name"`
+	Config               RETLSourceConfig `json:"config"`
+	SourceType           string           `json:"sourceType"`
+	SourceDefinitionName string           `json:"sourceDefinitionName"`
+	AccountID            string           `json:"accountId"`
+}
+
+type RETLSourceUpdateRequest struct {
+	ID        string           `json:"id,omitempty"`
+	Name      string           `json:"name"`
+	Config    RETLSourceConfig `json:"config"`
+	IsEnabled bool             `json:"enabled,omitempty"`
+	AccountID string           `json:"accountId"`
+	SourceID  string           `json:"sourceId"`
+}
+
+// RETLSourceConfig represents the config of a RETL SQL model source
+type RETLSourceConfig struct {
```

Comment:
Let's call this `RETLSQLModelConfig` for now. We don't have to support other source types for now, and when we do can probably introduce other create/update functions that expect different inputs for tables, profiles etc. It's better though if we don't name this as a generic source config type, to not misinterpret this as the common struct for all possible config schemas.